File: TwoPointer/ThreeSumSmaller.java
import java.util.Arrays;

public class ThreeSumSmaller {
    public static void main(String[] args) {
        ThreeSumSmaller solution = new ThreeSumSmaller();
        int[] nums = { -2, 0, 1, 3 };
        int target = 2;
        int ans = solution.input(nums, target);
        System.out.println(ans);
    }

    public int input(int[] nums, int target) {
        Arrays.sort(nums);
        int count = 0;
        for (int i = 0; i < nums.length; i++) {
            int l = i + 1;
            int r = nums.length - 1;
            while (l < r) {
                int sum = nums[i] + nums[l] + nums[r];
                if (sum < target) {
                    count += r - l;
                    l++;
                } else {
                    r--;
                }
            }
        }
        return count;
    }
}

File: TwoPointer/ThreeSumSmaller.js
var threeSumSmaller=function(nums,target){
    nums.sort((a,b)=>a-b);
    let count=0;
    for(let i=0;i<nums.length-2;i++){
        let left=i+1;
        let right=nums.length-1;
        while(left<right){
            if(nums[i]+nums[left]+nums[right]<target){
                count +=right-left;
                left++;
            }else{
                right--;
            }
        }
    }
return count;
};
const nums=[-2,0,1,3];
const target=2;
const result =threeSumSmaller(nums, target);
console.log(result);